onnxruntime/core/providers/cuda/tensor/split_impl.cu

```diff
@@ -157,5 +157,122 @@ Status SplitImpl(cudaStream_t stream, const size_t element_size, const int block
+  auto input_v = reinterpret_cast<size_t>(input);
+  auto out_v = reinterpret_cast<size_t>(out);
+  // select the min alignment of input and output pointer
+  auto vec_size = std::min(select(input_v), select(out_v));
+
+  if (vec_size == 16) {
+    _vec_copy_data<int4>(out, input, copy_size_in_byte);
+  } else if (vec_size == 8) {
+    _vec_copy_data<int2>(out, input, copy_size_in_byte);
+  } else if (vec_size == 4) {
+    _vec_copy_data<int32_t>(out, input, copy_size_in_byte);
+  } else if (vec_size == 2) {
+    _vec_copy_data<int16_t>(out, input, copy_size_in_byte);
+  } else {
+    _vec_copy_data<char>(out, input, copy_size_in_byte);
+  }
+}
+
+
+template <typename T>
+__global__ void _Split3InnerKernel(const int64_t size0,
+                                   const int64_t size1,
+                                   const int64_t size2,
+                                   const T* input_data,
+                                   T* output_data0,
+                                   T* output_data1,
+                                   T* output_data2,
+                                   const int64_t outer_size,
+                                   const int64_t inner_size) {
+  // each block copy a slice row of input data
+  int64_t row_id = blockIdx.x / 3;
+  int64_t slice_id = blockIdx.x % 3;
+#define _SELECT_1_in_3(slice_id, s1, s2, s3) \
+  ((slice_id == 0) ? (s1) : ((slice_id == 1) ? (s2) : (s3)))
+
+  // move input pointer to the corresponding row, and based on the slice_id, move to the begin of the slice
+  auto input_ptr = input_data + row_id * inner_size + _SELECT_1_in_3(slice_id, 0, size0, size0+size1);
+
+  auto output_ptr = _SELECT_1_in_3(slice_id,
+                                  output_data0 + row_id * size0,
+                                  output_data1 + row_id * size1,
+                                  output_data2 + row_id * size2);
+  auto copy_size = _SELECT_1_in_3(slice_id, size0, size1, size2);
+
+  _block_copy_data(output_ptr, input_ptr, copy_size);
+#undef _SELECT_1_in_3
+}
+
+Status Split3Inner(cudaStream_t stream, const size_t element_size, const int64_t size0, const int64_t size1,
+                   const int64_t size2, const void* input_data, void* output_data0, void* output_data1,
+                   void* output_data2, const gsl::span<const int64_t>& input_shape) {
+  CUDA_LONG outer_size = 1;
+  for (size_t i = 0; i < input_shape.size() - 1; ++i) {
+      outer_size *= static_cast<CUDA_LONG>(input_shape[i]);
+  }
+  CUDA_LONG inner_size = static_cast<CUDA_LONG>(input_shape[input_shape.size() - 1]);
+
+  int blocksPerGrid = outer_size * 3;
+
+  switch (element_size) {
+#define CASE_ELEMENT_TYPE(type)                                                                 \
+  case sizeof(type): {                                                                          \
+    _Split3InnerKernel<<<blocksPerGrid, kNumThreadsPerBlock, 0, stream>>>(size0, size1, size2,   \
```

Comment:
See [maxThreadsPerBlock](https://github.com/microsoft/onnxruntime/blob/3a917e49fb09e0e6542370bfbcf13d64c4115d1e/onnxruntime/core/providers/cuda/tensor/split_impl.cu#L18)

I would suggest to stick with 256.